### PR TITLE
[Improvement][service] Refactor service module to fix code smell

### PR DESCRIPTION
**Pull request by Segun-Ogundipe**
## What is the purpose of the pull request

This pull request fixes some of the code smell and vulnerability reported by sonar
Fixes part of #3325 

## Brief change log

  - *Refactor files to fix code smell and vulnerability reported by sonar in the service module*

## Verify this pull request

This pull request is code cleanup without any test coverage.

### CODE REVIEW

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessService.java [hunk 1]
```diff
@@ -916,12 +912,10 @@ private String joinGlobalParams(String parentGlobalParams, String subGlobalParam
      */
     private void initTaskInstance(TaskInstance taskInstance) {
 
-        if (!taskInstance.isSubProcess()) {
-            if (taskInstance.getState().typeIsCancel() || taskInstance.getState().typeIsFailure()) {
-                taskInstance.setFlag(Flag.NO);
-                updateTaskInstance(taskInstance);
-                return;
-            }
+        if (!taskInstance.isSubProcess() && taskInstance.getState().typeIsCancel() || taskInstance.getState().typeIsFailure()) {
```
Comment (dailidong) [inline dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessService.java:915]: this should be changed like this:
   if (!taskInstance.isSubProcess() && (taskInstance.getState().typeIsCancel() || taskInstance.getState().typeIsFailure())) {

how do you think?

Comment (Segun-Ogundipe) [reply dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessService.java:915]: That's true. I missed it

Comment (Segun-Ogundipe) [reply dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessService.java]: Done

**Comment by chengshiwen** [comment]
@Segun-Ogundipe  You should pass the **SonarCloud Code Analysis** at first.

**Comment by Segun-Ogundipe** [comment]
@chengshiwen, I'll write the tests necessary to do that

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessService.java [hunk 2]
```diff
@@ -508,16 +506,14 @@ public void createRecoveryWaitingThreadCommand(Command originCommand, ProcessIns
...
      * @param cmdParam cmdParam map
      * @return date
      */
     private Date getScheduleTime(Command command, Map<String, String> cmdParam) {
         Date scheduleTime = command.getScheduleTime();
-        if (scheduleTime == null) {
-            if (cmdParam != null && cmdParam.containsKey(CMDPARAM_COMPLEMENT_DATA_START_DATE)) {
-                scheduleTime = DateUtils.stringToDate(cmdParam.get(CMDPARAM_COMPLEMENT_DATA_START_DATE));
-            }
+        if (scheduleTime == null && cmdParam != null && cmdParam.containsKey(CMDPARAM_COMPLEMENT_DATA_START_DATE)) {
```
Comment (lenboo) [inline dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessService.java:515]: three judgement is complicated.
the schedulerTime can be judged alone.

**Review by dailidong — APPROVED** [review]
+1
good job
thanks for your contribution

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessService.java (hunk 2 above)
Comment (dailidong) [reply dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessService.java:515]: this is ok

**Pull request merged**